### PR TITLE
Update pylint

### LINKED ISSUE #43: Update CNES Pylint extension and Pylint versions
Docker CAT should use up-to-date tools for Python.

Consider updating Pylint and its CNES extension:
- Pylint version 2.5.0
- CNES Pylint extension 5.0.0

### PR DESCRIPTION
**Pull request by ThomasCluzel**
## Proposed changes

> _Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue._

Update of Pylint from version 1.5 to 2.5.0 and cnes-pylint-extension from 1.0 to 5.0.0

As a consequence, Python 2 is replaced by Python 3 and pip is used to manage the dependencies of cnes-pylint-extension.

By the way, CppCheck is now built from sources since its package for the requested version is no longer available on the debian repository causing the image build to fail.

## Types of changes

What types of changes does your code introduce to this project?
> _Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)

## Issues closed by changes

> _List here all issues closed by your changes. Use a list of items like `- [x] Close #0`_

- [x] Close #43 

## Checklist

> _Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I agree with the [CODE OF CONDUCT](CODE_OF_CONDUCT.md)
- [ ] Lint and unit tests pass locally with my changes
- [ ] SonarCloud and Travis CI tests pass with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -151,10 +146,17 @@ RUN echo 'deb http://ftp.fr.debian.org/debian/ bullseye main contrib non-free' >
     && ./rats \
     && cd .. \
     && rm -rf ./rats-2.4.tgz ./rats-2.4 \
+    && tar -zxvf cppcheck-1.90.tar.gz \
+    && cd cppcheck-1.90/ \
+    && make install MATCHCOMPILER="yes" FILESDIR="/usr/share/cppcheck" HAVE_RULES="yes" CXXFLAGS="-O2 -DNDEBUG -Wall -Wno-sign-compare -Wno-unused-function -Wno-deprecated-declarations" \
```
Comment (Sancretor) [inline Dockerfile:151]: Don't we need to do `configure` and/or `make` before `make install` ?

File: README.md [hunk 2]
```diff
@@ -116,15 +116,20 @@ You can run an analysis with the classic method by using one of scanners provide
...
+| pytest-runner                                         | `5.2`                                                 |
+| wrapt                                                 | `1.12.1`                                              |
+| six                                                   | `1.14.0`                                              |
+| lazy-object-proxy                                     | `1.4.3`                                               |
+| mccabe                                                | `0.6.1`                                               |
+| isort                                                 | `4.3.21`                                              |
+| typed-ast                                             | `1.4.1`                                               |
+| astroid                                               | `2.4.0`                                               |
+| pylint                                                | `2.5.0`                                               |
+| cnes-pylint-extension                                 | `5.0.0`                                               |
```
Comment (Sancretor) [inline README.md:132]: Maybe we should add the `:new:` icon to indicate what changed.
Of course, we will need to remove this icon from tools that did not changed like SonarQube, Cppcheck, etc.

File: Dockerfile (hunk 1 above)
Comment (ThomasCluzel) [reply Dockerfile:151]: No, we don't.
See the official [README](https://github.com/danmar/cppcheck/tree/1.90#gnu-make), autotools is not even supported but a wide range of build systems and build management systems are.

File: README.md (hunk 2 above)
Comment (ThomasCluzel) [reply README.md:132]: Good idea :+1:
I'm doing it right away.

File: Dockerfile (hunk 1 above)
Comment (Sancretor) [reply Dockerfile:151]: Thanks for the answer, and the link to the documentation.
By the way, as a reviewer, I should be the one to resolve this conversation...

File: README.md [hunk 3]
```diff
@@ -105,37 +105,42 @@ You can run an analysis with the classic method by using one of scanners provide
...
+| setuptools                                            | `46.1.3`                                              |
+| setuptools-scm                                        | `3.5.0`                                               |
+| pytest-runner                                         | `5.2`                                                 |
+| wrapt                                                 | `1.12.1`                                              |
+| six                                                   | `1.14.0`                                              |
+| lazy-object-proxy                                     | `1.4.3`                                               |
+| mccabe                                                | `0.6.1`                                               |
+| isort                                                 | `4.3.21`                                              |
+| typed-ast                                             | `1.4.1`                                               |
+| astroid                                               | `2.4.0`                                               |
```
[the remark this answers is not in the document]
Comment (ThomasCluzel) [reply README.md:130]: Well, yes they are.
Actually, the only reason why they are present is that they are dependencies of Pylint.
Anyway, they are new so they deserve the :new: icon.

**Pull request merged**